### DISCUSSION TITLE
[Native DLL] Array of strings

### DISCUSSION THREAD

**Discussion by ghost** [pascalabcnet/pascalabcnet#3011] [General]
C declaration: `int Init(int argc, char *argv[])`.

My PascalABC.NET translation: `function Init(argc: integer; argv: array of string): integer; external 'mydll.dll' name 'Init';`

Currently, I call it like this: `Init(0, nil);` (because I don't know how to get `argc` and `argv` of PascalABC.NET program).

But, I don't know if my translation is right or not, too. I also don't know how to pass the value of `argv`.

**Comment by SunSerega** [comment]
I don't think starting a separate discussion was warranted, isn't it the same thing continued?

Well, anyway, if you need to pass arguments and not the function - you have `System.Environment.GetCommandLineArgs`.
Though I'm not sure how the default marshaler would handle an `array of string`, you might need to use the `System.Runtime.InteropServices.Marshal` class manually...

Also, this much is easily googleable, since it's common for all .Net languages. You'll move forward much faster if you'll learn to find info yourself, instead of asking every little thing.

**Reply by ghost** [reply]
I googled, but I'm unsure.
